### ISSUE TITLE
Immediate disconnected(7) after running /honeywell

### ISSUE THREAD

**Issue by ramgolams** [fusterjj/HoneywellSecurityMQTT#8]
As soon as I run honeywell it shows a disconnect mqtt message. I tried running with and without a username and password and it's the same problem. I've tested my mqtt broker with another client and there are no issues there.

![image](https://user-images.githubusercontent.com/48026739/53431293-3a86e800-39be-11e9-8174-fe3bf106694b.png)

**Comment by fusterjj** [comment]
What broker are you using?  I've almost exclusively used Mosquitto.  Does it support MQTT v3.1.1?

**Comment by ramgolams** [comment]
I'm using mosquitto as well.

https://www.home-assistant.io/addons/mosquitto/

**Comment by jlt24** [comment]
I have this same problem.  Did you find a solution?
NM - it was a problem with my mqtt_congif.h

**Comment by ramgolams** [comment]
I never found a solution. How did you fix it?

**Comment by curt7000** [comment]
Encountering the same problem.  Does anyone know how to resolve this?  Running Mosquitto w/ MQTT Ver. 3.1.1 and credentials are confirmed working with other applications.

**Comment by curt7000** [comment]
> I never found a solution. How did you fix it?

Figured it out!

If your MQTT Broker requires username/password authentication, add the reference line below to your mqtt_config.h file, in Line 2, of the mqtt.cpp file, then rebuild.

#include "mqtt_config.h"

Step #1 - nano mqtt.cpp
Step #2 - Add to Line 2, #include "mqtt_config.h"
Step #3 - Rebuild using the command, ./build.sh

You should then be able to immediately connect to your MQTT Broker with username/password authentication.

**Comment by ramgolams** [comment]
Do you need to do this even if using anonymous auth?

**Comment by curt7000** [comment]
No.  This is only if your MQTT Broker requires username/password.
